lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -610,7 +645,7 @@ def internal_check_illegal_assignment(o)
     else
       # recursively check all contents unless it's a lambda expression. A lambda may contain
       # local assignments
-      o.eContents.each {|model| internal_check_illegal_assignment(model) } unless o.is_a?(Model::LambdaExpression)
+      o._pcontents {|model| internal_check_illegal_assignment(model) } unless o.is_a?(Model::LambdaExpression)
```

Comment:
Hm, the `_pcontent` is a somewhat strange name. Good to start with `_`, but is a `p` enough to denote pcore operations? I can imagine `_pcore_contents`, or `_pc_contents`  (to save three chars). IIRC there are not that many mehtods now (maybe just the `_pcontents`, but if we continue with others maybe it is best to include the full `pcore` in the name?